rotkehlchen/chain/evm/decoding/pendle/decoder.py

```diff
@@ -0,0 +1,94 @@
+
+
+NEW_LOCK_POSITION_TOPIC: Final = b'\xb1\xa37\x19V\xc5M\xc1\xd86\x95\xb4\xa0\x06\xb0Q\xc81>\xe9\x86\xe53\xb6\xb9d\xe7|\x90f\xfc,'  # noqa: E501
+WITHDRAW_TOPIC: Final = b'\x0e\x1b\xb0T\\\x1e\xbb\x9f\xb6\x80\xbd\xe75\x14\xe5r\x83\x1d\xe9;G\x9c\x08~\xc1\xefl5\xc3\xa1\x9f\xd6'  # noqa: E501
+
+
+class PendleCommonDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            pendle_token: 'Asset',
+            ve_pendle_contract: ChecksumEvmAddress,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.pendle_token = pendle_token
+        self.ve_pendle_contract = ve_pendle_contract
+
+    def _decode_ve_pendle_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == NEW_LOCK_POSITION_TOPIC:
+            for event in context.decoded_events:
+                # we don't compare the amount because it is the total amount locked that is emitted
+                # and not the amount that was currently deposited to be locked.
+                if (
+                        event.address == self.ve_pendle_contract and
+                        event.asset == self.pendle_token and
+                        event.event_type == HistoryEventType.SPEND and
+                        event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.notes = f'Lock {event.amount} PENDLE in voting escrow until {timestamp_to_date((lock_time := deserialize_timestamp(int.from_bytes(context.tx_log.data[32:64]))), formatstr="%d/%m/%Y")}'  # noqa: E501
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.extra_data = {'lock_time': lock_time}
+                    event.counterparty = CPT_PENDLE
+                    break
+
+        elif context.tx_log.topics[0] == WITHDRAW_TOPIC:
+            amount = token_normalized_value_decimals(
+                token_amount=int.from_bytes(context.tx_log.data[:32]),
+                token_decimals=DEFAULT_TOKEN_DECIMALS,
+            )
+            for event in context.decoded_events:
+                if (
+                    event.amount == amount and
+                    event.address == self.ve_pendle_contract and
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.counterparty = CPT_PENDLE
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.notes = f'Withdraw {amount} PENDLE from vote escrow'
```

Comment:
you didn't break here. I guess it is unique so stop the loop once you find the event